In [1]:
#Finds matches for search term in a page
from bs4 import BeautifulSoup
import requests
import re
import sys
import urllib
from urllib.request import Request, urlopen

def search_url(url, search_term):
  
  # print(url.split("www.")[-1].split("//")[-1].split(".")[0])
  # html = urllib.urlopen(url).read()
  #print (url)
  link=url
  if sys.version_info[0] == 3:
      from urllib.request import urlopen
  else:
      # Not Python 3 - today, it is most likely to be Python 2
      # But note that this might need an update when Python 4
      # might be around one day
      from urllib import urlopen

  try:  
    url=Request(url, headers={'User-Agent': 'Mozilla/5.0}'})
    # Your code where you can use urlopen
    with urlopen(url) as url:
        s = url.read()
  except:
    return 0
  # print(s)

  soup = BeautifulSoup(s, 'html.parser')

  # kill all script and style elements
  for script in soup(["script", "style"]):
      script.extract()    # rip it out

  # get text
  text = soup.get_text()

  # break into lines and remove leading and trailing space on each
  lines = (line.strip() for line in text.splitlines())
  # break multi-headlines into a line each
  chunks = (phrase.strip() for line in lines for phrase in line.split("  "))
  # drop blank lines
  text = '\n'.join(chunk for chunk in chunks if chunk)

  # print(text)
  search_term=search_term.lower()
  #text=text.lower()
  matches=0
  full_extract=""
  matched_sentences=[]

  for line in text.split("\n"):
    if len(line.split(" "))>20:
      for sentence in line.split('.'):
        for match in [m.start() for m in re.finditer(search_term, sentence.lower())]:
            #print (sentence+".")
            matched_sentences.append(sentence+".")
            #print()
            matches=matches+1        
        
  print("Matches in this url - "+str(matches))   
  return({
      "url":link,
      "matches":matches,
      "matched_sentences":matched_sentences
  })

# url="https://www.indiatoday.in/india/story/india-china-military-level-talks-ladakh-depsang-situation-1709067-2020-08-08"
# search_term="Ladakh"
# search_url(url, search_term)

In [2]:
#Reruens list of URLs found in a page
def get_urls(url):
    all_url_count=0
    url_count=0
    url_list=[]

    try:
        r  = requests.get(url)
        data = r.text
        soup = BeautifulSoup(data)    

        for link in soup.find_all('a'):
            all_url_count=all_url_count+1
        #[(all_url_count=all_url_count+1) for link in soup.find_all('a')]
        print("All URL count is "+str(all_url_count))

        # regex = r"(?i)\b((?:https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'\".,<>?«»“”‘’]))"
        regex=r'^(http|https|www)'
        for link in soup.find_all('a'):
            url=link.get('href')
            if (url) and bool(re.search(regex, url)):
                #print(url)
                url_list.append(url)
                url_count=url_count+1
            else:
                #print("This was NOT COUNTED "+str(url))
                pass
    except:
        pass
        
            
    print("Filtered URL count is "+str(url_count))
    
    url_list = list(dict.fromkeys(url_list))    
    #print(url_list)
    print("Duplicates removed URL count is "+str(len(url_list)))
    return(url_list)

In [ ]:
from bs4 import BeautifulSoup
import requests
import re
import time
import datetime
start_time=datetime.datetime.now()
print(start_time)

# url = raw_input("Enter a website to extract the URL's from: ")
# url="https://medium.com/@dmahugh_70618/deploying-a-flask-app-to-google-app-engine-faa883b5ffab"
# url="https://matadornetwork.com/notebook/how-to-travel-by-cargo-ship/"
# url="https://www.thehindubusinessline.com/opinion/columns/r-srinivasan/india-incs-living-in-denial/article32278690.ece?__sta=vhg.uosvpxezzlbsksaq%7CFYU&__stm_medium=email&__stm_source=smartech&utm_source=ThePrint&utm_medium=Email&utm_campaign=TFPMailer&utm_content=8Aug2020"

# source_url="https://www.indiatoday.in/india/story/india-china-military-level-talks-ladakh-depsang-situation-1709067-2020-08-08" 
# search_term="ladakh"

# source_url="https://en.wikipedia.org/wiki/Tom_Crean_(explorer)"
# search_term="antarctica"

source_url="https://en.wikipedia.org/wiki/Central_Intelligence_Agency"
search_term="cuba"

# source_url="https://science.sciencemag.org/content/359/6380/1146"
# search_term="virality"


depth=2
print("Depth set as "+str(depth))
articles_searched=0
links_parsed=[]

def search_driver(source_url, level):
    level=level+1
    if level<=depth:
        print("    "*level+"LEVEL is "+str(level))    
        print("SOURCE URL IS "+source_url)

        match_dict=search_url(source_url, search_term)
        if match_dict!=0 and match_dict["matches"]!=0:
            print(match_dict)    
        if (level+1<=depth): #So that we don't waste time extracting URLs that are beyond required depth
            for url in get_urls(source_url):
                global links_parsed
                if url not in links_parsed: #Remove lnks that have already been parsed earlier
                    #print("in loop level is "+str(level)+" "+url)
                    links_parsed.append(url)
                    global articles_searched
                    articles_searched=articles_searched+1
                    print()
                    search_driver(url,level)

                    
search_driver(source_url, -1)
print("\nArticles_searched "+str(articles_searched))
print("Articles_searched "+str(len(links_parsed)))

end_time=datetime.datetime.now()
print(end_time)
print("TIME taken in seconds is "+str(end_time-start_time))

2020-08-10 01:59:13.510691
Depth set as 2
LEVEL is 0
SOURCE URL IS https://en.wikipedia.org/wiki/Central_Intelligence_Agency
Matches in this url - 10
{'url': 'https://en.wikipedia.org/wiki/Central_Intelligence_Agency', 'matches': 10, 'matched_sentences': ['In the election of Patrice Lumumba, and his acceptance of Soviet support the CIA saw another possible Cuba.', 'The Bay of Pigs Invasion was a failed military invasion of Cuba undertaken by the CIA-sponsored paramilitary group Brigade 2506 on April 17, 1961.', ' Launched from Guatemala, the invading force was defeated within three days by the Cuban Revolutionary Armed Forces, under the direct command of Prime Minister Fidel Castro.', ' The CIA proceeded to organize the operation with the aid of various Cuban counter-revolutionary forces, training Brigade 2506 in Guatemala.', ' Over 1,400 paramilitaries set out for Cuba by boat on April 13.', ' Two days later on April 15, eight CIA-supplied B-26 bombers attacked Cuban airfields.', ' Th

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.wikidata.org/wiki/Special:EntityPage/Q37230
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://stats.wikimedia.org/#/en.wikipedia.org
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS http://www.cia.gov
Matches in this url - 0
All URL count is 229
Filtered URL count is 18
Duplicates removed URL count is 14

        LEVEL is 2
SOURCE URL IS https://youtu.be/KvG8c8aVtl8
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/CIA
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/Central.Intelligence.Agency
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.youtube.com/user/ciagov
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.flickr.com/photos/ciagov
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.youtube.com/user/ciagov
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://subscribe.washingtonpost.com/acq?promo=o2_gift_wpfd20&destination=https://www.washingtonpost.com/world/national-security/black-budget-summary-details-us-spy-networks-successes-failures-and-objectives/2013/08/29/7e57bb78-10ab-11e3-8cdd-bcdc09410972_story.html&tid=nav_giftsub_logged_out
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://subscribe.washingtonpost.com/acquisition/?nid=top_pb_subscribe&promo=o3&oscode=RPWH&acqEntType=mktg_onsite_article&destination=https://www.washingtonpost.com/world/national-security/black-budget-summary-details-us-spy-networks-successes-failures-and-objectives/2013/08/29/7e57bb78-10ab-11e3-8cdd-bcdc09410972_story.html&tid=nav_subscribe_logged_out
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.washingtonpost.com/subscribe/signin/?tid=nav_sign_in
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.washingtonpost.com/my-post/#/?tid=nav_ac

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.fas.org/siteindx.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.fas.org/search.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.fas.org/join.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.fas.org/sgp/foia/hogan.html
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS https://fas.org/blog/secrecy/2011/10/brennan_ctr_report.html
Matches in this url - 0
All URL count is 76
Filtered URL count is 58
Duplicates removed URL count is 43

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/fascientists
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/FAScientists
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.linkedin.com/groups/Federation-American-Scientists-FAS-3024349
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/about-fas/
Matches in this url - 0

        LEV

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/travel
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/health
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/home
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/opinion
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-srv/weather
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/weekly

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/digest
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/admin/classifieds
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/print
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-adv/a

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://newsvote.bbc.co.uk/email/news

        LEVEL is 2
SOURCE URL IS http://news.bbc.co.uk/1/hi/help/4735697.stm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.bbcworldservice.com/podcasts
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://news.bbc.co.uk/1/hi/help/3281849.stm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://news.bbc.co.uk/1/hi/help/default.stm

        LEVEL is 2
SOURCE URL IS http://www.bbc.co.uk/privacy/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://news.bbc.co.uk/1/hi/help/3281815.stm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.bbc.co.uk/info/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://news.bbc.co.uk/newswatch/ukfs/hi/feedback/default.stm

    LEVEL is 1
SOURCE URL IS https://www.theguardian.com/world/2006/oct/23/mainsection.tomphillips
Matches in this url - 1
{'url': 'https://www.theguardia

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://theguardian.newspapers.com
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://discountcode.theguardian.com
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://puzzles.theguardian.com/download
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/mobile/2014/may/29/the-guardian-for-mobile-and-tablet
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/membership
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/weekly?INTCMP=gdnwb_mawns_editorial_gweekly_GW_TopNav_Int
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/world/world
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/dialog/share?app_id=202314643182694&href=https%3A%2F%2Fwww.theguardian.com%2Fworld%2F2006%2Foct%2F23%2Fmainsection.tomphillips&CMP=share_btn_fb
Matches

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/world/report-ties-100-plus-cyber-attacks-on-us-computers-to-chinese-military/2013/02/19/2700228e-7a6a-11e2-9a75-dab0201670da_story.html
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS https://www.cia.gov/about-cia/leadership
Matches in this url - 0
All URL count is 214
Filtered URL count is 19
Duplicates removed URL count is 14

        LEVEL is 2
SOURCE URL IS http://www.youtube.com/watch?v=3UpC2IMg65k&feature=share&list=PLb9QRVXYyLdo7IUvybsKrJh4V-1b8NaX7
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://flic.kr/ps/UdHY1
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS https://www.nbcnews.com/news/other/sisterhood-spies-women-crack-code-cia-f2D11594601
Matches in this url - 0
All URL count is 118
Filtered URL count is 100
Duplicates removed URL count is 76

        LEVEL is 2
SOURCE URL IS https://www.nbcnews.com
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS ht

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://coupons.today.com
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.facebook.com/nbcnews
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.twitter.com/nbcnews
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://secure.nbcnews.com/_tps/accounts/newsletters
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.today.com/video/today/53552129
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/sharer/sharer.php?u=https://www.today.com/video/today/53552129
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/intent/tweet?text=Ann Curry: CIA now nearly half female&via=nbcnews&url=https://www.today.com/video/today/53552129&original_referer=https://www.today.com/video/today/53552129

        LEVEL is 2
SOURCE URL IS https://www.nbcnews.com/newsletters/breakingnews
Matches in this url - 0

        LEVEL is 

Matches in this url - 0
All URL count is 0
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://www.cia.gov/library/reports/general-reports-1/gulfwar/061997/support.htm
Matches in this url - 0
All URL count is 204
Filtered URL count is 19
Duplicates removed URL count is 14

    LEVEL is 1
SOURCE URL IS https://web.archive.org/web/20150314073648/https://www.cia.gov/about-cia/leadership/ciaorgchart.jpg/image.jpg
Matches in this url - 0
All URL count is 23
Filtered URL count is 14
Duplicates removed URL count is 12

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140724183745/https://www.cia.gov/about-cia/leadership/ciaorgchart.jpg/image.jpg
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20150414091329/https://www.cia.gov/about-cia/leadership/ciaorgchart.jpg/image.jpg
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20131017135446/https://www.cia.gov/about-

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://scoopnewsgroup.com/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.addtoany.com/add_to/twitter?linkurl=https%3A%2F%2Fwww.fedscoop.com%2Fnew-cia-director-inherits-agency-quickly-developing-cyber-capabilities%2F&linkname=New%20CIA%20director%20inherits%20an%20agency%20that%20is%20quickly%20developing%20cyber%20capabilities
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.addtoany.com/add_to/facebook?linkurl=https%3A%2F%2Fwww.fedscoop.com%2Fnew-cia-director-inherits-agency-quickly-developing-cyber-capabilities%2F&linkname=New%20CIA%20director%20inherits%20an%20agency%20that%20is%20quickly%20developing%20cyber%20capabilities
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.addtoany.com/add_to/linkedin?linkurl=https%3A%2F%2Fwww.fedscoop.com%2Fnew-cia-director-inherits-agency-quickly-developing-cyber-capabilities%2F&linkname=New%20CIA%20director%20inherits%20an%20

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Finterc.pt%2F2mCQ2aC
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/intent/tweet?text=CIA%E2%80%99s%20new%20%E2%80%9Cdigital%20innovation%E2%80%9D%20division%20can%E2%80%99t%20seem%20to%20keep%20its%20own%20secrets%20https%3A%2F%2Finterc.pt%2F2mCQ2aC%20by%20%40Schwartzesque
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.cia.gov/about-cia/leadership/Org_Chart_Oct2015.pdf


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://nsarchive.files.wordpress.com/2015/10/ncs.pdf


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.cia.gov/news-information/press-releases-statements/2015-press-releases-statements/message-to-workforce-agencys-blueprint-for-the-future.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://wikileaks.org/ciav7p1/cms/page_20251099.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bloomberg.com/news/articles/2016-08-01/cia-cyber-official-sees-data-flood-as-both-godsend-and-danger
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.nytimes.com/2013/05/21/us/politics/white-house-defends-tracking-fox-reporter.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.ap.org/ap-in-the-news/2013/govt-obtains-wide-ap-phone-records-in-probe
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.washingtonpost.com/world/national-security/why-its-hard-to-draw-a-line-between-snowden-and-the-paris-attacks/2015/11/18/34793ad4-8e28-11e5-baf4-b

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fcw.com/articles/2020/07/17/williams-army-it-reorg-crawford.aspx
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fcw.com/blogs/fcw-insider/2020/08/aug07topstories.aspx
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fcw.com/articles/2020/08/06/williams-nsa-microsoft-cloud.aspx
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fcw.com/articles/2020/07/17/johnson-irs-audit.aspx
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://pubads.g.doubleclick.net/gampad/jump?iu=/5978/eof.fcw&t=item%253d2d013f08_b54b_4746_89f2_a011b41b74df%26pos%253dBOX_6%26Topic%253dIntelligence_Agencies%252cInnovation%252cFederal%252cAgencies%252cTechnology%252cSectors&sz=300x250|1x1&tile=9&c=123456789
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://pubads.g.doubleclick.net/gampad/jump?iu=/5978/eof.fcw&t=item%253d2d013f08_b54b_4746_89f2_a011b41b74df%26pos%253dCUSTOM

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/gramm-leach-bliley-act-c-84
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/hipaa-hitech-c-282
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/cobit-c-441
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/pci-c-295
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/security-operations-c-444
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/ddos-protection-c-359
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/virtualization-cloud-security-c-445
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/identity-access-management-c-446
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/blogs/attackers-iot-paradise-billions-insecure-devices-p-2922
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/using-machine-learning-to-fight-money-laundering-a-14782
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/building-stronger-security-infrastructure-a-14775
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/pandemic-impact-how-will-data-breaches-evolve-a-14741
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/webinars?reset=true
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/course-library?reset=true
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/webinars-calendar
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankin

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.linkedin.com/company/information-security-media-group-corp-ismg/?viewAsMember=true

        LEVEL is 2
SOURCE URL IS https://events.ismg.io/event/cybersecurity-summit-new-york-city-ny-2020/?rf=trending
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/webinars/live-webinar-building-secure-convenient-customer-journey-convenience-w-2526?rf=trending
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/governance-risk-management-c-93
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/authors/eric-chabrow-i-249
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.twitter.com/GovInfoSecurity
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bankinfosecurity.com/cia-creates-digital-innovation-directorate-a-7991#disqus_thread
Matches in this url - 0

        LEVEL is 2
SO

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/podcasts/510317/its-been-a-minute-with-sam-sanders
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/podcasts/510312/codeswitch
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/podcasts/510333/throughline
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/podcasts/510355/considerthis
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://ondemand.npr.org/anon.npr-mp3/npr/me/2004/05/20040528_me_02.mp3?orgId=1&topicId=1003&p=3&story=1914175&dl=1&siteplayer=true&dl=1


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/sections/national/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/programs/morning-edition/2004/05/28/12923919/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/people/2780701/mary-louise-kelly
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.twitter.com/NPRKelly
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/NPR/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/NPR
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.instagram.com/npr/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.npr.org/contact

        LEVEL is 2
SOURCE URL IS https://help.npr.org

        LEVEL is 2
SOURCE URL IS https://shop.npr.org/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.nprpresents.org
Matches in this url - 0

        LE

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/07/nrac-nrl/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/07/hollywood-and-the-bomb/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/07/jason-labs/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/07/military-deaths-trends/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/security/2020/06/the-state-departments-compliance-report-plays-the-blame-game-despite-offering-little-evidence/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/06/credibility-crisis/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/06/pidb-vision/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://fas.org/blogs/secrecy/2020/06/retiree-recall/
Matches in this u

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 6
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS http://www2.gwu.edu/~nsarchiv/NSAEBB/NSAEBB434/
Matches in this url - 1
{'url': 'http://www2.gwu.edu/~nsarchiv/NSAEBB/NSAEBB434/', 'matches': 1, 'matched_sentences': ['was not used; System XII was a radar warner developed during the Cuba Missile Crisis; and Systems XIII, XIV and XV were adaptations of Navy ECM devices.']}
All URL count is 67
Filtered URL count is 47
Duplicates removed URL count is 34

        LEVEL is 2
SOURCE URL IS https://nsarchive2.gwu.edu/index.html
Matches in this url - 1
{'url': 'https://nsarchive2.gwu.edu/index.html', 'matches': 1, 'matched_sentences': [' government documents, photographs, and more relating to Area 51, the Cuban Missile Crisis and more in this publication of the Carnegie Corporation of New York.']}

        LEVEL is 2
SOURCE URL IS https://nsarchive.gwu.edu/about
Matches in this url - 2
{'url': 'https://nsarchive.gw

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://nsarchive2.gwu.edu/NSAEBB/NSAEBB186/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://nsarchive2.gwu.edu/NSAEBB/NSAEBB74/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://nsarchive2.gwu.edu/NSAEBB/NSAEBB35/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://blog.foreignpolicy.com/posts/2013/08/15/declassified_the_cias_secret_history_of_area_51#.Ug1V_hBr9Vw.twitter
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.cnbc.com/id/100968502
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.today.com/video/today/52771502#52771502
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.usatoday.com/story/news/nation/2013/08/16/area-51-cia/2663829/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.thedailybeast.com/cheats/2013/08/16/cia-admits-to-secret-area-51.html
Matches in this url - 0

        LEVEL is 2
SOUR

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 1
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://fas.org/sgp/foia/intel98.html
Matches in this url - 0
All URL count is 15
Filtered URL count is 10
Duplicates removed URL count is 5

    LEVEL is 1
SOURCE URL IS https://www.nytimes.com/2005/11/08/politics/08budget.html
Matches in this url - 0
All URL count is 32
Filtered URL count is 21
Duplicates removed URL count is 19

        LEVEL is 2
SOURCE URL IS https://www.nytimes.com/section/us
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/dialog/feed?app_id=9869919170&link=https%3A%2F%2Fwww.nytimes.com%2F2005%2F11%2F08%2Fpolitics%2Fofficial-reveals-budget-for-us-intelligence.html%3Fsmid%3Dfb-share&name=Official%20Reveals%20Budget%20for%20U.S.%20Intelligence&redirect_uri=https%3A%2F%2Fwww.facebook.com%2F
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.n

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18833#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18832#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18828#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18826#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18825#respond
Matches in this

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18717#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18684#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18682#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18676#respond
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/exclusive-dozens-of-cia-operatives-on-the-ground-during-benghazi-attack/?replytocom=18675#respond
Matches in this

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://twitter.com/TheLeadCNN
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/theleadcnn
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/01/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/02/
Matches in this url - 4
{'url': 'https://thelead.blogs.cnn.com/2013/08/02/', 'matches': 4, 'matched_sentences': ['Miami (CNN)\xa0- All Luis had to do was hand a Cuban birth certificate to immigration officials and he was on his way to becoming a U.', 'Those are the privileges afforded to Cubans who flee the Castro regime and make it to the United States to seek asylum.', " He's Venezuelan, not Cuban.", ' Pretending to be Cuban has become a trend in visa fraud, according to Alysa Erichs, special agent in charge of theU.']}

        LEVEL is 2
SOURCE URL IS https://thelead.blogs.cnn.com/2013/08/05/
Matches in th

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 0
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS http://www.spiegel.de/politik/deutschland/verbindungsstelle-61-staatsanwaelte-ermitteln-gegen-hohen-bnd-mann-a-882145.html
Matches in this url - 0
All URL count is 254
Filtered URL count is 236
Duplicates removed URL count is 192

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/politik/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://abo.spiegel.de/?b=SPOHNAVABO&requestAccessToken=true&sara_icid=disp_upd_9h6L5hu8K1AAnttzYATx3hvk7taDkP&targetUrl=https%3A%2F%2Fwww.spiegel.de%2Ffuermich%2F
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://gruppenkonto.spiegel.de/authenticate?requestAccessToken=true&targetUrl=https%3A%2F%2Fwww.spiegel.de%2Ffuermich%2F
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/fuermich

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/gesundheit/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/familie/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/psychologie/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/karriere/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/deinspiegel/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.bento.de/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.manager-magazin.de/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/video/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/thema/index-a/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/tests/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/backstage/
Matc

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/thema/cia/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/kontakt?betreff=Ermittlungen+gegen+Chef+von+geheimer+BND-Gruppe&channel=Politik&id=5aa85f7d-0001-0001-0000-000000882145
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://spiele.spiegel.de
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.spiegel.de/netzwelt/spiele/trivial-pursuit-auf-spiegel-online-was-wissen-sie-wirklich-von-der-welt-a-1292479.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://spiele.spiegel.de/exchange/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://spiele.spiegel.de/bubble-shooter/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://spiele.spiegel.de/sudoken/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://spiele.spiegel.de/fibonacci/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS


        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://www.thestar.ie/star/video-kathryn-thomas-gets-wet-and-wild-for-ice-bucket-challenge-51741/

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://www.thestar.ie/star/lifes-a-pitch-51732/

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://www.thestar.ie/star/cilla-black-absolutely-positive-cliff-richard-sex-claims-are-false-51736/

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://www.thestar.ie/star/sweet-maria-crowned-rose-of-tralee-51731/

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://www.thestar.ie/star/joy-of-six-hundred-grand-for-utvs-pat-kenny-51722/

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://www.thestar.ie/star/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140820222128/http://thestar.

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.facebook.com/time
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.twitter.com/time
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.pinterest.com/timemagazine/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://timemagazine.tumblr.com/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://cloud.newsletters.time.com/newsletters?source=time_nav/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://cgi.timeinc.net/cgi-bin/mail/dnp/privacy_centralized.cgi/time?dnp_source=E

        LEVEL is 2
SOURCE URL IS https://subscription.timeinc.com/storefront/privacy/time/generic_privacy_new.html?dnp-source=B#california
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://cgi.timeinc.net/cgi-bin/mail/dnp/terms_of_service.cgi/time

        LEVEL is 2
SOURCE URL IS http://subscription-assets.timeinc.com/prod/assets/themes/magazines/default/tem

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 20
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://www.dni.gov/index.php/ic-legal-reference-book/national-security-act-of-1947
Matches in this url - 4
{'url': 'https://www.dni.gov/index.php/ic-legal-reference-book/national-security-act-of-1947', 'matches': 4, 'matched_sentences': ['\xa0\xa0\xa0\xa0\xa0 \xa0 Summary of intelligence relating to terrorist recidivism of detainees held at United States Naval Station, Guantanamo Bay, Cuba.', ' \xa0 SUMMARY OF INTELLIGENCE RELATING TO TERRORIST RECIDIVISM OF DETAINEES HELD AT UNITED STATES NAVAL STATION, GUANTANAMO BAY, CUBA \xa0 SEC.', '—The Director of National Intelligence, in consultation with the Director of the Central Intelligence Agency and the Director of the Defense Intelligence Agency, shall make publicly available an unclassified summary of— (1) intelligence relating to recidivism of detainees currently or formerly held at the Naval Detention 

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 5
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://history.state.gov/historicaldocuments/frus1945-50Intel
Matches in this url - 0
All URL count is 109
Filtered URL count is 10
Duplicates removed URL count is 10

        LEVEL is 2
SOURCE URL IS https://static.history.state.gov/frus/frus1945-50Intel/ebook/frus1945-50Intel.epub


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://static.history.state.gov/frus/frus1945-50Intel/ebook/frus1945-50Intel.mobi


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://bookstore.gpo.gov/catalog/foreign-relations-united-states-series-frus
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.worldcat.org/search?q=isbn%3A0160452082
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.worldcat.org/search?q=isbn%3A9780160452086
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.state.gov/misc/207091.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.state.gov/misc/415.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.state.gov/misc/152386.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.state.gov/misc/87529.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.state.gov
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS http://www.foia.cia.gov/sites/default/files/document_conversions/44/2010-05-01.pdf


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 0
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://www.cia.gov/library/center-for-the-study-of-intelligence/kent-csi/vol38no1/pdf/v38i1a06p.pdf


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 0
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://www.cia.gov/news-information/featured-story-archive/2014-featured-story-archive/a-look-back-the-first-director-of-central-intelligence.html
Matches in this url - 0
All URL count is 209
Filtered URL count is 12
Duplicates removed URL count is 12

    LEVEL is 1
SOURCE URL IS https://history.state.gov/historicaldocuments/frus1945-50Intel/d292
Matches in this url - 0
All URL count is 85
Filtered URL count is 5
Duplicates removed URL count is 5

    LEVEL is 1
SOURCE URL IS https://www.cia.gov/library/center-for-the-study-of-intelligence/csi-publications/csi-studies/studies/fall_winter_2001/article06.html
Matches in this url - 0
All URL count is 335
Filtered URL count is 19
Duplicates removed URL count is 14

    LEVEL is 1
SOURCE URL IS https://books.google.com/books?id=Zh1VtsxRlRAC
Matches in this url - 0
All URL count is 190
Filtered URL count is 175

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Committee&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=communism&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=communist&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=concerning&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Consul&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Council&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=c

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Office&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=once&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Party&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Philippines&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=Plan&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=policymakers&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Zh1VtsxRlRAC&q=political

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=bibliogroup:%22American+Studies%22&source=gbs_metadata_r&cad=6
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Frances+Gouda%22&source=gbs_metadata_r&cad=6
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Thijs+Brocades+Zaalberg%22&source=gbs_metadata_r&cad=6
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Political+Science%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Political+Science+International+Relations%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Political+Science+International+Relations+General%22
Matches in this url - 0



Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.nytimes.com/subscribe/help/plugins.html

        LEVEL is 2
SOURCE URL IS https://www.nytimes.com/library/world/mideast/041600iran-cia-intro.html
Matches in this url - 2
{'url': 'https://www.nytimes.com/library/world/mideast/041600iran-cia-intro.html', 'matches': 2, 'matched_sentences': ['the disastrous Cuban intervention known as the Bay of Pigs in 1961.', '"From time to time," he continued, "I gave talks on the operation to various groups within the agency, and, in hindsight, one might wonder why no one from the Cuban desk ever came or read the history.']}

        LEVEL is 2
SOURCE URL IS https://www.nytimes.com/library/world/mideast/041600iran-cia-chapter1.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.nytimes.com/library/world/mideast/041600iran-cia-chapter2.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.nytimes.com/library/world/mideast/041600iran-cia-chap

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=7vcGTI4Q22YC&printsec=frontcover&source=gbs_vpt_read
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=7vcGTI4Q22YC&sitesec=buy&source=gbs_vpt_read
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.rowmanlittlefield.com/isbn/0842024751
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://books.rediff.com/book/ISBN:0842025901
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.flipkart.com/search.php?query=0842025901

        LEVEL is 2
SOURCE URL IS http://www.infibeam.com/Books/search?q=9780842025904
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/url?id=7vcGTI4Q22YC&pg=PA2&q=http://worldcat.org/oclc/1026110150&clientid=librarylink&usg=AFQjCNHo6tolMeyGOJesIl0Q3k8y31iCzQ&source=gbs_buy_r
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.c

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Odd+Arne+Westad%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Professor+of+International+History+Odd+Arne+Westad%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Mg9UN_YjovkC&source=gbs_citations_module_r&cad=3
Matches in this url - 3
{'url': 'https://books.google.co.in/books?id=Mg9UN_YjovkC&source=gbs_citations_module_r&cad=3', 'matches': 3, 'matched_sentences': [' In a compelling and provocative history that takes readers to fourteen countries, including Cuba, Iran, South Vietnam, Chile, and Iraq, Kinzer surveys modern American history from a new and often surprising perspective.', " Read full reviewSelected pagesTitle PageTable of ContentsIndexReferencesContentsIntroduction 1 Despotism and Godless Terrorism 111 Get Rid of This Stinker 129 Not the

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Nick+Cullather%22&source=gbs_metadata_r&cad=5
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+Latin+America%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+Latin+America+Central+America%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+/+Americas+(North,+Central,+South,+West+Indies)%22&source=gbs_metadata_r&cad=5
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+/+General%22&source=gbs_metadata_r&cad=5
Matches in this url - 0

        LEVEL

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=Castillo+Armas&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=Central&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=close&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=Colonel&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=Communism&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=communist&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=Panama&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=party&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=peasants&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=Peurifoy&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=plot&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=political&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=mS7ZVKa6i3AC&q=position&so

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Victor+Bulmer-Thomas%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=fHYWAAAAYAAJ&source=gbs_citations_module_r&cad=5
Matches in this url - 2
{'url': 'https://books.google.co.in/books?id=fHYWAAAAYAAJ&source=gbs_citations_module_r&cad=5', 'matches': 2, 'matched_sentences': [' She employs a number of case studies of intervention and analyzes decision-making patterns from the early years of the cold war in Guatemala and Cuba to the post-cold-war policies in Panama and the war on drugs in Peru.', ' CottamNo preview available - 1994Common terms and phrasesaccepted achieve actions administration affect approach argued associated behavior believed Bush Carter Castro cause Central America cold war communist complex concern Congress considered containment continued coup Cuba decision democratic Department dependent image developed

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=bibliogroup:%22Princeton+Paperbacks%22&source=gbs_metadata_r&cad=6
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Piero+Gleijeses%22&source=gbs_metadata_r&cad=6
Matches in this url - 7
{'url': 'https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Piero+Gleijeses%22&source=gbs_metadata_r&cad=6', 'matches': 7, 'matched_sentences': ['in › booksPiero Gleijeses uses archival sources, particularly from the United States, South Africa, and the closed Cuban archives, to provide an unprecedented international history of this important theater of the late Cold War.', 'in › booksThis is a compelling and dramatic account of Cuban policy in Africa from 1959 to 1976 and of its escalating clash with U.', "The Cuban Drumbeat: Castro's Worldview : Cuban Foreign .", "The Cuban Drumbeat: Castro's Worldview : Cuban Foreign .

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&printsec=frontcover&source=gbs_atb
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&sitesec=buy&source=gbs_atb
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.ohioswallow.com/book/Managing+the+Counterrevolution
Matches in this url - 2
{'url': 'http://www.ohioswallow.com/book/Managing+the+Counterrevolution', 'matches': 2, 'matched_sentences': ['For forty years the Cuban Revolution has been at the forefront of American public opinion, yet few are knowledgeable about the history of its enemies and the responsibility of the U.', ' government in organizing and sustaining the Cuban counterrevolution.']}

        LEVEL is 2
SOURCE URL IS http://books.rediff.com/book/ISBN:0896802159
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.flipkart.com/search.php?query=0896802159

        LEVEL is 2


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=Armas%E2%80%99s&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=assistance&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=August&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=Carlos+Castillo+Armas&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=Central+America&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=CIA%E2%80%99s&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.c

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=loan&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=lots+58&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=mala&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=malan&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=Mallory&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=March&source=gbs_word_cloud_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=h17R_A0n-1MC&q=memcon&source=gbs_w

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=X65lUn9YK38C&source=gbs_citations_module_r&cad=5
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=X65lUn9YK38C&printsec=references&vq=%22Managing+the+Counterrevolution%22&source=gbs_citations_module_r&cad=5
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22J.+Patrice+McSherry%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?q=cites:ISBN0896802159&id=h17R_A0n-1MC&source=gbs_citations_module_r&cad=5
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://chart.googleapis.com/chart?chs=400x400&cht=qr&chl=https://books.google.co.in/books?id=h17R_A0n-1MC&source=qrcode


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=bibliogroup:%22Latin+America+series%5D%22&source=gbs_metadata_r&cad=6
Matches in this url - 2
{'url': 'https://www.google.co.in/search?tbo=p&tbm=bks&q=bibliogroup:%22Latin+America+series%5D%22&source=gbs_metadata_r&cad=6', 'matches': 2, 'matched_sentences': ['The Cuban Counterrevolutionhttps://books.', 'in › booksArboleya also analyzes the role played by Cuban immigrants to the United States and the perspectives for improvement in relations between the two nations as a result of the generational and social changes that have been occurring in the .']}

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22%5BResearch+in+international+studies%22&source=gbs_metadata_r&cad=6
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=bibliogroup:%22Ohio+RIS+Latin+America+Series%22&source=gbs_metadata_r

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://scholar.google.com/scholar?hl=en&q=+%0A+1954i+Central+Intelligence+Bulletin%2C+%28June+27%29+top+secret%2C+declassified+1977+in+DDRS+%281977%2C+89E%3A+6%29
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://scholar.google.com/scholar_lookup?hl=en&publication_year=1965&author=+Chancellor%2C+John&title=The+Science+of+Spying
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://scholar.google.com/scholar_lookup?hl=en&publication_year=1976&author=+Cline%2C+Ray&title=Secrets%2C+Spies+and+Scholars
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://scholar.google.com/scholar_lookup?hl=en&publication_year=1978&author=+Colby%2C+William&title=Honorable+Men%3A+My+Life+in+the+CIA
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://scholar.google.com/scholar_lookup?hl=en&publication_year=1981&author=+Cook%2C+Blanche&title=The+Declassified+Eisenhower
Matches in this url - 0

        L

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.deepdyve.com/doc-view?docId=10.1177/0094582X8301000106&fieldName=0094-582X&affiliateId=sage
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://journals.sagepub.com/doi/10.1177/0095327X8401000202?icid=int.sj-abstract.similar-articles.1
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://journals.sagepub.com/doi/10.1177/0094582X8301000103?icid=int.sj-abstract.similar-articles.2
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://journals.sagepub.com/doi/10.1177/0094582X8301000111?icid=int.sj-abstract.similar-articles.3
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://dx.doi.org/10.1080/17449359.2018.1431551
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://doi.org/10.1080/17449359.2018.1431551
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://dx.doi.org/10.1007/978-3-658-16235-1_8
Matches in this url - 3
{'url': 'http://dx.doi.org

Matches in this url - 2
{'url': 'https://books.google.co.in/url?id=vf9ZJx8WkjQC&q=http://www.librarything.com/work/3374621/reviews/173665766&usg=AFQjCNEnaI-9ESaR-gUzUFRAo72COrSySg&source=gbs_site_section_reviews_ge_r', 'matches': 2, 'matched_sentences': [" It has not provided good intelligence to policy makers; it missed the mark on Castro and Cuba, failed in understanding eastern Europe and the Soviet Union, got it completely wrong about WMD's in Iraq and more.", ' To appreciate what Weiner has to say, it helps to know the basics about the overthrow of governments in Iran and Guatemala, the Bay of Pigs fiasco, the Cuban Missile Crisis, the U2 incident, the Kennedy assassination and the assassination of Ngo Dinh Diem, Iran Contra, Noriega in Panama, the kidnapping and bombing of Americans in Lebanon, Gulbuddin Hekmatyar in Afghanistan, Mogadishu, the bombing of the Cole, the African embassy bombings, the bombing of the aspirin plant in Sudan, American support of the Baath party in Iraq

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Tim+Weiner%22&source=gbs_metadata_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+Military%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+Military+United+States%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+/+Military+/+United+States%22&source=gbs_metadata_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22History+/+United+States+/+General%22&source=gbs_metadata_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Political+Science+/+Security+(National+%26+International)%22&source=gb

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/content/world/index.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/content/world/wires/index.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.washingtonpost.com/wp-dyn/content/article/2007/06/27/AR2007062700190_pf.html
Matches in this url - 3
{'url': 'http://www.washingtonpost.com/wp-dyn/content/article/2007/06/27/AR2007062700190_pf.html', 'matches': 3, 'matched_sentences': ["And while Cuban government press officials didn't return a call seeking reaction Tuesday, the release of the newly declassified CIA documents had already been noted in state media.", " government's 10-most wanted list: Momo Giancana, Al Capone's successor in Chicago; and Santos Trafficante, one of the most powerful mobsters in Batista's Cuba.", 'This particular assassination attempt was dropped after the failed CIA-sponsored Bay of Pigs invasio

Matches in this url - 2
{'url': 'https://archive.org/details/spywhosavedworld00jerr', 'matches': 2, 'matched_sentences': [' victory during the Cuban missile crisis.', ' The Spy Who Saved the World makes a vital contribution to our understanding of the ramifications and ultimate meaning of the Cold War and provides a fresh perspective an the fragmentation of the Soviet Union now reaching its climaxIncludes bibliographical references (pages 443-473) and indexAcknowledgments -- Transliteration of Russian names -- Prologue -- Approaching the Americans -- The British connection -- London at last -- Life and times -- Hitting the jackpot -- Penkovsky\'s travels in England -- Trading secrets -- Meeting a Cabaret girl and "C" -- Return to Moscow -- Return to London -- Paris -- Safety or glory -- Closing the missile gap -- Suspicion and surveillance -- End game: The Cuban missile crisis -- The trial -- Aftermath -- Traitor or savior?.']}
All URL count is 76
Filtered URL count is 33
Duplicates re

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0
All URL count is 49
Filtered URL count is 0
Duplicates removed URL count is 0

    LEVEL is 1
SOURCE URL IS https://history.state.gov/historicaldocuments/frus1964-68v30/d342
Matches in this url - 0
All URL count is 91
Filtered URL count is 5
Duplicates removed URL count is 5

    LEVEL is 1
SOURCE URL IS https://archive.org/details/warofnumbersinte00adam
Matches in this url - 0
All URL count is 72
Filtered URL count is 36
Duplicates removed URL count is 32

        LEVEL is 2
SOURCE URL IS https://twitter.com/intent/tweet?url=https://archive.org/details/warofnumbersinte00adam&via=internetarchive&text=War+of+numbers+%3A+Adams%2C+Sam+%3A+Free+Download%2C+Borrow%2C+and+Streaming+%3A+Internet+Archive
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/sharer/sharer.php?u=https://archive.org/details/warofnumbersinte00adam
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.reddit.com/submit?url=https://archive.org/detai


        LEVEL is 2
SOURCE URL IS http://www.nytimes.com/2000/02/11/opinion/an-african-pinochet.html
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.amnesty.org/fr/library/asset/AFR20/004/2001/fr/2343f1a0-d902-11dd-ad8c-f3d4445c118e/afr200042001en.html

        LEVEL is 2
SOURCE URL IS http://www.usip.org/sites/default/files/file/resources/collections/commissions/Chad-Report.pdf


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.reagan.utexas.edu/archives/speeches/1985/70885a.htm

        LEVEL is 2
SOURCE URL IS https://foreignpolicy.com/wp-content/themes/foreign-policy-2017/features/assets/images/our-man-in-africa/inline7.jpg


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://foreignpolicy.com/wp-content/themes/foreign-policy-2017/features/assets/images/our-man-in-africa/inline8.jpg


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://foreignpolicy.com/wp-content/themes/foreign-policy-2017/features/assets/images/our-man-in-africa/inline9.jpg


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://foreignpolicy.com/wp-content/themes/foreign-policy-2017/features/assets/images/our-man-in-africa/inline10.jpg


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://foreignpolicy.com/wp-content/themes/foreign-policy-2017/features/assets/images/our-man-in-africa/inline11.jpg


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://foreignpolicy.com/wp-content/themes/foreign-policy-2017/features/assets/images/our-man-in-africa/inline12.jpg


Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.hrw.org/news/2012/03/09/chronology-habr-case
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.icj-cij.org/docket/files/144/17064.pdf

        LEVEL is 2
SOURCE URL IS http://www.state.gov/j/gcj/us_releases/reports/2012/193222.htm

        LEVEL is 2
SOURCE URL IS https://wpvip.com/?utm_source=vip_powered_wpcom&utm_medium=web&utm_campaign=VIP%20Footer%20Credit&utm_term=foreignpolicy.com
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS https://www.theguardian.com/world/2013/jul/03/chad-hissene-habre-charged
Matches in this url - 0
All URL count is 150
Filtered URL count is 44
Duplicates removed URL count is 34

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/world/series/guardian-africa-network
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/world/chad
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.facebook.com/dialog/shar

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140302090727/http://www.thenation.com/?utm_source=main_nav_home&utm_medium=website&utm_campaign=main_nav_home
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140302090727/http://www.thenation.com/columns?utm_source=main_nav_columns&utm_medium=website&utm_campaign=main_nav_columns
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140302090727/http://www.thenation.com/archive
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140302090727/http://www.thenation.com/multimedia?utm_source=main_nav_multimedia&utm_medium=website&utm_campaign=main_nav_multimedia
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20140302090727/http://nationbuilders.thenation.com/?utm_source=main_nav_nation_builders&utm_medium=website&utm_campaign=main_nav_nation_builder
Matches

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/travels
Matches in this url - 3
{'url': 'https://www.thenation.com/travels', 'matches': 3, 'matched_sentences': ['We started our land excursions in 2014, when, after 16 years of successful cruises, The Nation applied for and was granted a special people-to-people license to host educational trips to Cuba, well before the Obama administration relaxed US travel restrictions to the island.', ' Since then, we’ve hosted almost 20 trips to Cuba, bringing close to 1,000\xa0people to the beautiful, fascinating country with an itinerary specially curated for politically sophisticated travelers.', "Street AddressStreet Address Line 2CityState / Province / RegionPostal / Zip CodeCountryAfghanistanAlbaniaAlgeriaAndorraAngolaAntigua and BarbudaArgentinaArmeniaAustraliaAustriaAzerbaijanBahamasBahrainBangladeshBarbadosBelarusBelgiumBelizeBeninBhutanBoliviaBosnia and HerzegovinaBotswanaBrazilBruneiBulgariaBurkina FasoB

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/politics/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/subject/economy/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/subject/society/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/admin-taxonomy/take-action/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/about-us-and-contact/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/advertise/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/community-guidelines/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.thenation.com/contact-us/
Matches in this url - 1
{'url': 'https://www.thenation.com/contact-us/', 'matches': 1, 'matched_sentences': ["Street AddressStreet Address Line 2CityState / Province / RegionPostal / Zip Co

Matches in this url - 0
All URL count is 65
Filtered URL count is 33
Duplicates removed URL count is 28

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20131020001933/http://www.bt.com.bn/analysis/2008/12/17/story_of_us_cia_and_taliban
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20141006124600/http://www.bt.com.bn/analysis/2008/12/17/story_of_us_cia_and_taliban
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20120701131421/http://www.bt.com.bn:80/analysis/2008/12/17/story_of_us_cia_and_taliban
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20150923195252/http://www.bt.com.bn/analysis/2008/12/17/story_of_us_cia_and_taliban
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20131205090713/http://bt.com.bn/supp/e-islamia
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20131205090713/htt


        LEVEL is 2
SOURCE URL IS http://www.paktoday.com/latest.htm

        LEVEL is 2
SOURCE URL IS http://www.afgha.com/index.php

        LEVEL is 2
SOURCE URL IS https://www.telegraph.co.uk/telegraphtv/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.telegraph.co.uk/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.telegraph.co.uk/?utm_source=TMG&utm_medium=footer_home&utm_campaign=footerhomelink
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://preview.telegraph.co.uk/news/uknews/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://preview.telegraph.co.uk/news/politics/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://preview.telegraph.co.uk/news/features/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://preview.telegraph.co.uk/news/wikileaks-files/
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://jobs.telegraph.co.uk/
Matches in this url - 0

        LEVEL is 2

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Hlk7BAAAQBAJ&printsec=frontcover&vq=%22although+Casey+initiated%22&source=gbs_vpt_read
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Hlk7BAAAQBAJ&sitesec=buy&source=gbs_vpt_read
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://books.rediff.com/book/ISBN:1469618516
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.flipkart.com/search.php?query=1469618516

        LEVEL is 2
SOURCE URL IS http://www.infibeam.com/Books/search?q=9781469618517
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/url?id=Hlk7BAAAQBAJ&pg=PA110&q=http://worldcat.org/oclc/1125377772&clientid=librarylink&usg=AFQjCNEzbUjrcyNGlZXQjetVX1-nGvYRTw&source=gbs_buy_r
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=Hlk7BAAAQBAJ&sitesec=buy&source=gbs_buy_r
Matches in this

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20080813084613/http://www.fbi.gov/libref/historic/famcases/ames/ames.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20080524120111/http://www.fbi.gov:80/libref/historic/famcases/ames/ames.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20080709010642/http://www.fbi.gov/libref/historic/famcases/ames/ames.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20070607215448/http://www.fbi.gov:80/libref/historic/famcases/ames/ames.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20090715042329/http://www.fbi.gov/libref/historic/famcases/ames/ames.htm
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://archive.org/details/web_iq
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://web.archive.org/web/20080611072859/https

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/news/podcasts
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/news/photos
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/news/espanol
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/news/apps
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/investigate/terrorism
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/investigate/counterintelligence
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/investigate/cyber
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/investigate/public-corruption
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/investigate/civil-rights
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.fbi.gov/investigate/organized-crime
Matches in this 

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=F2nSTwbFAh0C&lpg=PA121&pg=PA119&redir_esc=y&hl=en&output=html_text
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.com/accounts/Login?service=print&continue=https://books.google.co.in/books%3Fop%3Dlookup%26id%3DF2nSTwbFAh0C%26continue%3Dhttps://books.google.co.in/books%253Fid%253DF2nSTwbFAh0C%2526lpg%253DPA121%2526pg%253DPA119%2526redir_esc%253Dy%2526hl%253Den&hl=en
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=F2nSTwbFAh0C&sitesec=buy&source=gbs_atb
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://books.rediff.com/book/ISBN:0061147788
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://www.flipkart.com/search.php?query=0061147788

        LEVEL is 2
SOURCE URL IS http://www.infibeam.com/Books/search?q=9780061147784
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.go

Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.


Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22George+Tenet%22&source=gbs_metadata_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=inauthor:%22Bill+Harlow%22&source=gbs_metadata_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Biography+%26+Autobiography%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Biography+%26+Autobiography+Political%22
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Biography+%26+Autobiography+/+Political%22&source=gbs_metadata_r&cad=4
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.co.in/search?tbo=p&tbm=bks&q=subject:%22Political+Science+/+General%22&source=gbs_metadata_r&cad=4
Matches

Matches in this url - 0

        LEVEL is 2
SOURCE URL IS http://books.google.co.in/support/answer/180577?hl=en-IN&url=https://books.google.co.in/books?id=F2nSTwbFAh0C&pg=PA121&q=%22Afghan+sanctuary%22&redir_esc=y&hl=en&v=F2nSTwbFAh0C&is=atb
Matches in this url - 0

    LEVEL is 1
SOURCE URL IS https://books.google.com/books?id=F2nSTwbFAh0C&pg=PA177
Matches in this url - 0
All URL count is 72
Filtered URL count is 58
Duplicates removed URL count is 47

        LEVEL is 2
SOURCE URL IS https://www.google.com/accounts/Login?service=print&continue=https://books.google.co.in/books%3Fid%3DF2nSTwbFAh0C%26pg%3DPA177%26redir_esc%3Dy%26hl%3Den&hl=en
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://books.google.co.in/books?id=F2nSTwbFAh0C&pg=PA177&redir_esc=y&hl=en&output=html_text
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://www.google.com/accounts/Login?service=print&continue=https://books.google.co.in/books%3Fop%3Dlookup%26id%3DF2nSTwbFAh0C%26continue%3Dh

Matches in this url - 0
All URL count is 160
Filtered URL count is 153
Duplicates removed URL count is 118

        LEVEL is 2
SOURCE URL IS https://www.theguardian.com/international
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://support.theguardian.com/contribute?INTCMP=header_support_contribute&acquisitionData=%7B%22componentType%22:%22ACQUISITIONS_HEADER%22,%22componentId%22:%22header_support_contribute%22,%22source%22:%22GUARDIAN_WEB%22%7D
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://support.theguardian.com/subscribe?INTCMP=header_support_subscribe&acquisitionData=%7B%22componentType%22:%22ACQUISITIONS_HEADER%22,%22componentId%22:%22header_support_subscribe%22,%22source%22:%22GUARDIAN_WEB%22%7D
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://jobs.theguardian.com/?INTCMP=jobs_int_web_newheader
Matches in this url - 0

        LEVEL is 2
SOURCE URL IS https://profile.theguardian.com/signin?INTCMP=DOTCOM_NEWHEADER_SIGNIN&compon

## OBSERVATIONS
### For IndiaToday
Around 15-20% URLs repeated on the page itself
source_url="https://www.indiatoday.in/india/story/india-china-military-level-talks-ladakh-depsang-situation-1709067-2020-08-08" 
search_term="ladakh"
* Depth=1, Time:40sec, Articles_searched:68;
* Depth=2, Time:47min, Articles_searched:3933(=68+3865);
* Depth=2, Time:40sec, Articles_searched:, With repetition check;

### For Wikipedia
source_url="https://en.wikipedia.org/wiki/Tom_Crean_(explorer)"
search_term="antarctica"
* Depth=1, Time:2min, Articles_searched:52;
* Depth=2, Time:min, Articles_searched:;

source_url="https://en.wikipedia.org/wiki/Central_Intelligence_Agency"
search_term="cuba"
Around 90% URLs repeated on the page itself
* Depth=1, Time16.5min:, Articles_searched:353;
* Depth=2, Time:, Articles_searched:;

### For Research paper
Around 85% URLs repeated on the page itself

source_url="https://en.wikipedia.org/wiki/Tom_Crean_(explorer)"
search_term="antarctica"
* Depth=1, Time:2.5min, Articles_searched:55;
* Depth=2, Time:1h20min, Articles_searched:684;

